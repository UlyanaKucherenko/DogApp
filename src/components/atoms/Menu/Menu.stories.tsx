import type { Meta, StoryObj } from '@storybook/react';

import { Menu } from '.';
import { menuHeader } from '../../../utils/const';

const meta = {
  title: 'Atoms/Menu',
  component: Menu,
  parameters: {
    layout: 'fullscreen'
  }
} satisfies Meta<typeof Menu>;

export default meta;
type Story = StoryObj<typeof meta>;

export const MenuNav: Story = {
  args: {
    menuList: menuHeader
  }
};
