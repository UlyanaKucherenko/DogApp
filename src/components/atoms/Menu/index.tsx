import { NavList, NavListItem } from './styled';

type MenuItemProps = {
  title: string;
  route: string;
};

export type MenuProps = {
  menuList: MenuItemProps[];
};

export const Menu = ({ menuList }: MenuProps) => (
  <NavList component="nav">
    {menuList.map(item => (
      <NavListItem to={item.route} key={item.title}>
        {({ isActive }) => (
          <span className={isActive ? 'active' : ''}>{item.title}</span>
        )}
      </NavListItem>
    ))}
  </NavList>
);
