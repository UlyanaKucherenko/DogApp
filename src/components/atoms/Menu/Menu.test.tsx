import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import * as stories from './Menu.stories';

describe('Menu', () => {
  const { MenuNav } = composeStories(stories);
  it('should match snapshot Menu', () => {
    const { container } = render(
      <BrowserRouter>
        <MenuNav />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
