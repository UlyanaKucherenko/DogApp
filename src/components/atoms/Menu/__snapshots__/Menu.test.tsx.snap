// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot Menu 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  max-width: 575px;
  width: 100%;
  padding: 0;
}

.emotion-1 {
  color: rgba(0, 0, 0, 0.87);
  background-color: transparent;
  font-size: 20px;
  border-radius: 6px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  white-space: nowrap;
  text-transform: capitalize;
  padding: 4px 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:hover {
  background-color: #1565c0;
  color: #fff;
}

.emotion-1:active {
  background-color: #1976d2;
  color: #fff;
  box-shadow: 3px 3px 0px 0px rgba(0, 0, 0, 1);
}

.emotion-1.active {
  background-color: #1976d2;
  color: #fff;
  box-shadow: 3px 3px 0px 0px rgba(0, 0, 0, 1);
}

<div>
  <nav
    class="MuiList-root MuiList-padding emotion-0"
  >
    <a
      aria-current="page"
      class="emotion-1 active"
      href="/"
    >
      <span
        class="active"
      >
        Main
      </span>
    </a>
    <a
      class="emotion-1"
      href="/breeds"
    >
      <span
        class=""
      >
        Breeds
      </span>
    </a>
    <a
      class="emotion-1"
      href="/vote"
    >
      <span
        class=""
      >
        Vote
      </span>
    </a>
    <a
      class="emotion-1"
      href="/history"
    >
      <span
        class=""
      >
        History
      </span>
    </a>
    <a
      class="emotion-1"
      href="/your-uploads"
    >
      <span
        class=""
      >
        Your uploads
      </span>
    </a>
  </nav>
</div>
`;
